platform_selector.py

```diff
@@ -0,0 +1,61 @@
+from selector import Selector
+from custom_dtypes import *
+# print(driver.platform_selected())
+
+# class Platform_Selector(Selector):
+class Platform_Selector(Selector):
+
+    def __init__(self, Repository):
+        self.__current_safety_monitor = SafetyMonitor.NO_SELECTION
+        self.__repo_image = Repository
+        self.__platform = SafetyMonitor.NO_SELECTION
+
+    def query_repository(self,):
+        self.__current_safety_monitor = self.__repo_image.get_active_safety_monitor()
+
+    def update_repository(self,):
+
+        self.__repo_image.update_current_safety_monitor(self.platform)
```

Comment:
self.__repo_image.update_platform_status has to be used. Wrong method. platform is a private attribute